distribution/lib/Standard/AWS/0.0.0-dev/src/S3.enso

```diff
@@ -0,0 +1,30 @@
+from Standard.Base import all
+
+import project.AWS_Credential.AWS_Credential
+import project.Errors.S3_Error
+import project.Internal.Auth
+
+polyglot java import software.amazon.awssdk.services.s3.S3Client
+polyglot java import software.amazon.awssdk.services.s3.model.S3Exception
+
+type S3
+    ## Gets the list of the S3 buckets.
+    @credentials AWS_Credential.default_widget
+    list_buckets : AWS_Credential | Nothing -> Vector Text ! S3_Error
+    list_buckets credentials:(AWS_Credential | Nothing)=Nothing = handle_s3_errors <|
+        client = make_client credentials
+        buckets = client.listBuckets.buckets
+        buckets.map b->b.name
+
+## PRIVATE
+handle_s3_errors : Function -> Any ! S3_Error
+handle_s3_errors  ~action =
+    Panic.catch S3Exception action caught_panic->
+        aws_error = caught_panic.payload.awsErrorDetails
+        Error.throw (S3_Error.Error aws_error.errorMessage aws_error.errorCode)
+
+## PRIVATE
+make_client : (AWS_Credential | Nothing) -> S3Client
+make_client credentials =
+    pvdr = Auth.create_provider credentials
+    S3Client.builder.credentialsProvider pvdr . build
```

Comment:
```suggestion
    provider = Auth.create_provider credentials
    S3Client.builder.credentialsProvider provider . build
```

I think better to use full names like this instead of shortcuts. At least these are our naming guidelines in style guides e.g. [for Java](https://enso.org/docs/developer/enso/style-guide/java.html#naming) (since we don't have one for Enso yet).

Reply:
(I don't want to be too nitpicky, I'm fine with single letter names like `p` if it is rather clear from the context what that is. But at the point that we use an abbreviation which is not much shorter from the proper word, it feels like making it slightly less readable for no real gain)